app/assets/javascripts/task_list.coffee

```diff
@@ -94,145 +87,174 @@
+      changedEvent = createEvent 'tasklist:changed',
+        index: index
+        checked: item.checked
+      @field.dispatchEvent changedEvent
+
+  # Static interface
+
+  @incomplete: "[ ]"
+  @complete: "[x]"
+
+  # Escapes the String for regular expression matching.
+  @escapePattern: (str) ->
+    str.
+      replace(/([\[\]])/g, "\\$1"). # escape square brackets
+      replace(/\s/, "\\s").         # match all white space
+      replace("x", "[xX]")          # match all cases
+
+  @incompletePattern: ///
+    #{@escapePattern(@incomplete)}
+  ///
+  @completePattern: ///
+    #{@escapePattern(@complete)}
+  ///
+
+  # Pattern used to identify all task list items.
+  # Useful when you need iterate over all items.
+  @itemPattern: ///
+    ^
+    (?:                     # prefix, consisting of
+      \s*                   # optional leading whitespace
+      (?:>\s*)*             # zero or more blockquotes
+      (?:[-+*]|(?:\d+\.))   # list item indicator
+    )
+    \s*                     # optional whitespace prefix
+    (                       # checkbox
+      #{@escapePattern(@complete)}|
+      #{@escapePattern(@incomplete)}
+    )
+    \s+                     # is followed by whitespace
+    (?!
+      \(.*?\)               # is not part of a [foo](url) link
+    )
+    (?=                     # and is followed by zero or more links
+      (?:\[.*?\]\s*(?:\[.*?\]|\(.*?\))\s*)*
+      (?:[^\[]|$)           # and either a non-link or the end of the string
+    )
+  ///
+
+  # Used to skip checkbox markup inside of code fences.
+  # http://rubular.com/r/TfCDNsy8x4
+  @startFencesPattern: /^`{3}.*$/
+  @endFencesPattern: /^`{3}$/
+
+  # Used to filter out potential mismatches (items not in lists).
+  # http://rubular.com/r/OInl6CiePy
+  @itemsInParasPattern: ///
+    ^
+    (
+      #{@escapePattern(@complete)}|
+      #{@escapePattern(@incomplete)}
+    )
+    .+
+    $
+  ///g
+
+  # Given the source text, updates the appropriate task list item to match the
+  # given checked value.
+  #
+  # Returns the updated String text.
+  @updateSource: (source, itemIndex, checked) ->
+    clean = source.replace(/\r/g, '').
+      replace(@itemsInParasPattern, '').
+      split("\n")
+    index = 0
+    inCodeBlock = false
+    result = for line in source.split("\n")
+      if inCodeBlock
+        # Lines inside of a code block are ignored.
+        if line.match(@endFencesPattern)
+          # Stop ignoring lines once the code block is closed.
+          inCodeBlock = false
+      else if line.match(@startFencesPattern)
+        # Start ignoring lines inside a code block.
+        inCodeBlock = true
+      else if line in clean && line.match(@itemPattern)
+        index += 1
+        if index == itemIndex
+          line =
+            if checked
+              line.replace(@incompletePattern, @complete)
+            else
+              line.replace(@completePattern, @incomplete)
+      line
+    result.join("\n")
+
+if window.jQuery
```

Comment:
Yep. This will allow projects to update to 2.x before they are ready to move away from jQuery.

I imagine this will be a lot easier to land in GitLab if we can drop it in with npm first, then use the vanilla javascript API in a separate MR.